python/cuml/tests/test_hdbscan.py

```diff
@@ -447,3 +461,116 @@ def test_hdbscan_plots():
+@pytest.mark.parametrize('cluster_selection_epsilon', [0.0, 0.5])
+@pytest.mark.parametrize('max_cluster_size', [0])
+@pytest.mark.parametrize('cluster_selection_method', ['eom', 'leaf'])
+def test_all_points_membership_vectors(nrows,
+                                       ncols,
+                                       nclusters,
+                                       cluster_selection_epsilon,
+                                       cluster_selection_method,
+                                       min_cluster_size,
+                                       allow_single_cluster,
+                                       max_cluster_size):
+    X, y = make_blobs(n_samples=nrows,
+                      n_features=ncols,
+                      centers=nclusters,
+                      cluster_std=0.7,
+                      shuffle=True,
+                      random_state=42)
+
+    cuml_agg = HDBSCAN(verbose=logger.level_info,
+                       allow_single_cluster=allow_single_cluster,
+                       max_cluster_size=max_cluster_size,
+                       min_cluster_size=min_cluster_size,
+                       cluster_selection_epsilon=cluster_selection_epsilon,
+                       cluster_selection_method=cluster_selection_method,
+                       prediction_data=True)
+    cuml_agg.fit(X)
+
+    sk_agg = hdbscan.HDBSCAN(
+        allow_single_cluster=allow_single_cluster,
+        approx_min_span_tree=False,
+        gen_min_span_tree=True,
+        min_cluster_size=min_cluster_size,
+        cluster_selection_epsilon=cluster_selection_epsilon,
+        cluster_selection_method=cluster_selection_method,
+        algorithm="generic",
+        prediction_data=True)
+
+    sk_agg.fit(cp.asnumpy(X))
+
+    cu_membership_vectors = all_points_membership_vectors(cuml_agg)
+    cu_membership_vectors.sort(axis=1)
+    sk_membership_vectors = hdbscan.all_points_membership_vectors(
+        sk_agg).astype("float32")
+    sk_membership_vectors.sort(axis=1)
+    assert_all_points_membership_vectors(
+        cu_membership_vectors,
+        sk_membership_vectors)
+
+
+@pytest.mark.parametrize('nrows', [1000, 10000])
+@pytest.mark.parametrize('n_points_to_predict', [200, 500])
+@pytest.mark.parametrize('ncols', [10, 25])
+@pytest.mark.parametrize('nclusters', [15])
+@pytest.mark.parametrize('min_cluster_size', [30, 60])
+@pytest.mark.parametrize('cluster_selection_epsilon', [0.0, 0.5])
+@pytest.mark.parametrize('max_cluster_size', [0])
+@pytest.mark.parametrize('cluster_selection_method', ['eom', 'leaf'])
+def test_approximate_predict(nrows,
+                             n_points_to_predict,
+                             ncols,
+                             nclusters,
+                             cluster_selection_epsilon,
+                             cluster_selection_method,
+                             min_cluster_size,
+                             max_cluster_size):
+    X, y = make_blobs(n_samples=nrows,
```

Comment:
I apologize that I missed this before, but we should test both approximate_predict and all_points_membership against multiple datasets. We have seen bugs arise from different data distributions. We want to do this in our automated tests to make sure we're battle tested